spec/openapi.yml

```diff
@@ -1041,55 +1061,19 @@ components:
           description:  The description of the dataset.
           type: string
         createdByRun:
-          description: The run associated with the dataset version.
-          type: object
-          properties:
-            id:
-              description: The ID of the run.
-              type: string
-              format: uuid
-            createdAt:
-              description: An [ISO-8601](https://en.wikipedia.org/wiki/ISO_8601) timestamp representing the date/time the run was created.
-              type: string
-              format: date-time
-            updatedAt:
-              description: An [ISO-8601](https://en.wikipedia.org/wiki/ISO_8601) timestamp representing the date/time the run was updated.
-              type: string
-              format: date-time
-            nominalStartTime:
-              description: An [ISO-8601](https://en.wikipedia.org/wiki/ISO_8601) timestamp representing the nominal start time of the run.
-              type: string
-              format: date-time
-            nominalEndTime:
-              description: An [ISO-8601](https://en.wikipedia.org/wiki/ISO_8601) timestamp representing the nominal end time of the run.
-              type: string
-              format: date-time
-            state:
-              description: The current state of the run.
-              type: string
-              enum: [ NEW, RUNNING, COMPLETED, FAILED, ABORTED ]
-            startedAt:
-              type: integer
-              description: The time at which the run was started. Measured in seconds since the Unix epoch.
-            endedAt:
-              type: integer
-              description: The time at which the run was ended. Measured in seconds since the Unix epoch.
-            durationMs:
-              type: integer
-              description: The total duration of the run. Measured in milliseconds.
-            args:
-              description: The arguments of the run.
-              type: object
-            jobVersion:
-              $ref: '#/components/schemas/jobVersionId'
-            inputVersions:
-              type: array
-              items:
-                $ref: '#/components/schemas/datasetVersionId'
-            outputVersions:
-              type: array
-              items:
-                $ref: '#/components/schemas/datasetVersionId'
+          $ref: '#/components/schemas/Run'
+        facets:
+          $ref: '#/components/schemas/DatasetFacets'
+      required:
```

Comment:
Do we need the `required` section in the dataset version response? Usually that to let the caller know what's required when making a request.

Reply:
For the code generation, filling out the `required` section lets the generator know whether to generate nullable types. E.g., in Typescript, the fields can be typed as `string | undefined`, which forces the user to check for nulls before using the field. In Java, the code can be generated with `@NonNull` annotations so we don't accidentally return a null value when a real value is expected.

Reply:
Ahh, very cool 👍. We'll want to update the other responses (can be in a follow up PR) to follow the same convention.